Template_Skill/src/main/java/com/wolfpack/database/DbConnection.java

```diff
@@ -0,0 +1,248 @@
+/**
+ * Before attempting to use this class, follow the directions on the wiki at
+ * https://github.com/370-Alexa-Project/CS370_Echo_Demo/wiki/Using-the-dbConnection-class
+ * to make sure you have everything set up properly. Examples of how to use this class
+ * are also provided.
+ * 
+ * This class can be used to simplify the process of connecting to a postgresql database.
+ * In order to connect to the database, you should call getCredentials() before calling
+ * getRemoteConnection(). Then you can run your queries with runQuery().
+ */
+
+package com.wolfpack.database;
+
+import java.util.*;
+
+import java.sql.*;
+
+import java.io.File;
+import org.w3c.dom.Document;
+
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.DocumentBuilder;
+import org.xml.sax.SAXException;
+import org.xml.sax.SAXParseException;
+
+public class DbConnection {
+
+	private static String dbName;
+	private static String username;
+	private static String password;
+	private static String hostName; // The url to the database
+	private static String port;
+	private static String localPathToSSL;
+	static Connection dbConnection;
+	private static boolean connected;
+
+	/**
+	 * When the main constructor is used, the object will attempt to get the
+	 * user's credentials and then use that to secure a connection to the
+	 * database.
+	 */
+	public DbConnection() {
+		connected = false;
+	}
+
+	/**
+	 * The second constructor allows you to pass the path to the credentials for
+	 * the database and then load in those credentials. This replaces calling
+	 * the main constructor and then needing to call getCredentials().
+	 */
+	public DbConnection(String pathToCredentials) {
+		connected = false;
```

Comment:
Normally you would call the other constructor here instead of duplicating the code.